sdk/python/kfp/dsl/component_factory.py

```diff
@@ -94,9 +95,11 @@ def make_index_url_options(pip_index_urls: Optional[List[str]]) -> str:
     index_url = pip_index_urls[0]
     extra_index_urls = pip_index_urls[1:]
 
-    options = [f'--index-url {index_url} --trusted-host {index_url}']
+    options = [
+        f'--index-url {index_url} --trusted-host {urllib.parse.urlparse(index_url).hostname}'
+    ]
     options.extend(
-        f'--extra-index-url {extra_index_url} --trusted-host {extra_index_url}'
+        f'--extra-index-url {extra_index_url} --trusted-host {urllib.parse.urlparse(extra_index_url).hostname}'
```

Comment:
Should we consider the `hostname` as good enough? No need to specify also the port in case it's present in the original URL? In other words - we trust all apps on that domain in such case, right?

Reply:
> Should we consider the hostname as good enough?

Yes, it is sufficient.

> No need to specify also the port in case it's present in the original URL?

Correct. The fix doesn't involve the port number.

> In other words - we trust all apps on that domain in such case, right?

Could you please rephrase your question?

Reply:
@jstourac please double-check https://github.com/kubeflow/pipelines/pull/11151